src/Visualizer/hooks.ts

```diff
@@ -131,21 +130,19 @@ export default function useHooks(
   }, []);
 
   // interaction mode
-  const [_interactionMode, changeInteractionMode] = useValue(
-    initialInteractionMode,
+  const [interactionMode, changeInteractionMode] = useValue(
+    initialInteractionMode || "default",
     onInteractionModeChange,
   );
-  const interactionMode = _interactionMode || "default";
 
-  const [cursor, setCursor] = useState<CursorType>("auto");
   useEffect(() => {
-    setCursor(
-      interactionMode === "sketch" ? "crosshair" : interactionMode === "move" ? "grab" : "auto",
-    );
+    if (interactionMode === "default") {
+      mapRef?.current?.engine?.setCursor("auto");
+    }
   }, [interactionMode]);
 
   // feature flags
-  const featureFlags = INTERACTION_MODES[interactionMode];
+  const featureFlags = INTERACTION_MODES[interactionMode ?? "default"];
```

Comment:
Is this necessary?
```suggestion
  const featureFlags = INTERACTION_MODES[interactionMode];
```

Reply:
It's coming from the type definition of `useValue` (T | undefined), although we provided the fallback value to avoid undefined in this case it still has the undefined in type.

Reply:
Ah I see. Thank you.